### ISSUE TITLE
Support amortized cost

### ISSUE THREAD

**Issue by cmeeren** [ChilliCream/graphql-platform#7403]
### Product

Hot Chocolate

### Is your feature request related to a problem?

I realize this is not part of the IBM cost spec, but I'd still like to air the issue.

Consider this schema, inferred from code where `MyThing.Value` is async:

```graphql
schema {
  query: Query
}

type MyThing {
  id: String
  value: String @cost(weight: "10")
}

type Query {
  test: [MyThing] @listSize(assumedSize: 100, requireOneSlicingArgument: false)
}
```

Now consider this query:

```graphql
query {
  test {
    id
    value
  }
}
```

As expected, this fails with the following response:

```json
{
  "errors": [
    {
      "message": "The maximum allowed field cost was exceeded.",
      "extensions": {
        "fieldCost": 1001,
        "maxFieldCost": 1000,
        "code": "HC0047"
      }
    }
  ]
}
```

However, imagine that `MyThing.Value` is using a dataloader. This means (at least as a first-order approximation) that the `Value` field cost of `10` should be counted only once, not multiplied by the assumed list size as it currently is.

I generally use data loaders for all async fields where I know they can be requested in a list, so this part of the cost spec is causing  some queries to have much too high costs.

### The solution you'd like

I'm open to suggestions and general thoughts on this. As for a concrete example solution, perhaps an `@amortizedCost` directive could be used similarly to `@cost`, but with the rule that it only applies its cost once in a list of items. (I have not thought long and hard about this, so it's quite possible there are much better solutions. I'm keen to hear them.)

**Comment by michaelstaib** [comment]
We will not add this as its not in the spec... but we are discussing how we should account for this or if its worth account for this.

**Comment by cmeeren** [comment]
Regardless of what comes out of your discussions, I'd be very interested in hearing the conclusion!

**Comment by cmeeren** [comment]
Hi! Did you get around to discussing this? I am very interested in hearing what you landed on and why.
